icechunk-python/tests/test_stateful_repo_ops.py

```diff
@@ -186,22 +173,108 @@ def checkout_branch(self, ref: str) -> None:
         self.checkout_commit(self.branches[ref])
         self.branch = ref
 
-    def reset_branch(self, branch, commit) -> None:
+    def reset_branch(self, branch: str, commit: str) -> None:
+        assert commit in self.commits
         self.branches[branch] = commit
 
     def delete_branch(self, branch_name: str) -> None:
         del self.branches[branch_name]
 
+    def delete_tag(self, tag: str) -> None:
+        del self.tags[tag]
+
     def create_tag(self, tag_name: str, commit_id: str) -> None:
+        assert commit_id in self.commits
         self.tags[tag_name] = TagModel(commit_id=str(commit_id))
+        self.created_tags.add(tag_name)
 
     def checkout_tag(self, ref: str) -> None:
         self.checkout_commit(self.tags[str(ref)].commit_id)
 
-    def list_prefix(self, prefix: str) -> None:
+    def list_prefix(self, prefix: str) -> tuple[str, ...]:
         assert prefix == ""
         return tuple(self.store)
 
+    def refs_iter(self):
+        tag_iter = map(operator.attrgetter("commit_id"), self.tags.values())
+        return itertools.chain(self.branches.values(), tag_iter)
+
+    def expire_snapshots(
```

Comment:
I think all of this is great, but, we are re-implementing the algorithms in the tests, potentially with the same bugs we are blind to. I wonder if in the future we can define properties of the GCed or Expired repo, and try to track less in the model.

Reply:
I landed up here because I'm doing an exact equality check for snapshot ids that were expired because only expires snaphsots that are reachable. 

I guess I can search the "forest" and assert that `set(IC expired IDs) < set(all possible expirable IDs)`